pytorch_lightning/plugins/training_type/ipu.py

```diff
@@ -129,10 +129,18 @@ def pre_dispatch(self) -> None:
         self._handle_gradient_accumulation_steps()
 
     @property
-    def replication_factor(self):
+    def replication_factor(self) -> int:
+        if not self.lightning_module:
+            # The plugin has been passed in by the user and has not been connected to the Trainer.
+            # Check if the user has passed in custom poptorch.Options to infer number of IPUs being used.
+            # In this scenario we prioritize the training options.
```

Comment:
cc @davors72 for stdout logging purposes, I default to the training opts replication factor. Obviously if this is setup different for inference and training it doesn't matter as this is just for logging purposes!